From a44c32f80087af14fa1a2863ac8a4053b4a2933b Mon Sep 17 00:00:00 2001
Message-Id: <a44c32f80087af14fa1a2863ac8a4053b4a2933b.1650636481.git.keller@fs-net.de>
In-Reply-To: <8876d3464f7abe86d96420dca69dd3317368c208.1650636480.git.keller@fs-net.de>
References: <8876d3464f7abe86d96420dca69dd3317368c208.1650636480.git.keller@fs-net.de>
From: Will Huang <wilhuang@codeaurora.org>
Date: Wed, 8 Jan 2020 15:21:26 +0800
Subject: [PATCH 09/56] qcacld-2.0: Keep pattern match configure same for all
 vdev

Currently ptrn_match_enable is saved per vdev interface, if change
pattern match configure through iwpriv command for one interface,
other vdev interface is not changed accordingly, it makes wow configure
not work as expected.

Change-Id: I470a1fec324d189d6dea6c4a7288c73d27947d39
CRs-Fixed: 2505478
---
 CORE/SERVICES/WMA/wma.c | 18 +++++++++++++-----
 1 file changed, 13 insertions(+), 5 deletions(-)

diff --git a/CORE/SERVICES/WMA/wma.c b/CORE/SERVICES/WMA/wma.c
index aeb93a1..1d864b4 100644
--- a/CORE/SERVICES/WMA/wma.c
+++ b/CORE/SERVICES/WMA/wma.c
@@ -27019,6 +27019,7 @@ static VOS_STATUS wma_wow_enter(tp_wma_handle wma,
 				tpSirHalWowlEnterParams info)
 {
 	struct wma_txrx_node *iface;
+	int i;
 
 	WMA_LOGD("wow enable req received for vdev id: %d", info->sessionId);
 
@@ -27028,9 +27029,12 @@ static VOS_STATUS wma_wow_enter(tp_wma_handle wma,
 		return VOS_STATUS_E_INVAL;
 	}
 
-	iface = &wma->interfaces[info->sessionId];
-	iface->ptrn_match_enable = info->ucPatternFilteringEnable ?
-							    TRUE : FALSE;
+	/* All interfaces must have same pattern match config */
+	for (i = 0; i < wma->max_bssid; i++) {
+		iface = &wma->interfaces[i];
+		iface->ptrn_match_enable = info->ucPatternFilteringEnable ?
+					   TRUE : FALSE;
+	}
 	wma->wow.magic_ptrn_enable = info->ucMagicPktEnable ? TRUE : FALSE;
 	wma->wow.deauth_enable = info->ucWowDeauthRcv ? TRUE : FALSE;
 	wma->wow.disassoc_enable = info->ucWowDeauthRcv ? TRUE : FALSE;
@@ -27046,6 +27050,7 @@ static VOS_STATUS wma_wow_exit(tp_wma_handle wma,
 			       tpSirHalWowlExitParams info)
 {
 	struct wma_txrx_node *iface;
+	int i;
 
 	WMA_LOGD("wow disable req received for vdev id: %d", info->sessionId);
 
@@ -27055,8 +27060,11 @@ static VOS_STATUS wma_wow_exit(tp_wma_handle wma,
 		return VOS_STATUS_E_INVAL;
 	}
 
-	iface = &wma->interfaces[info->sessionId];
-	iface->ptrn_match_enable = FALSE;
+	/* All interfaces must have same pattern match config */
+	for (i = 0; i < wma->max_bssid; i++) {
+		iface = &wma->interfaces[i];
+		iface->ptrn_match_enable = FALSE;
+	}
 	wma->wow.magic_ptrn_enable = FALSE;
 	vos_mem_free(info);
 
-- 
2.14.5

